Include/internal/pycore_bitutils.h

```diff
@@ -82,6 +84,31 @@ _Py_bswap64(uint64_t word)
 }
 
 
+// Population count: count the number of 1's in 'u'
+// (number of bits set to 1)
+static inline int
+_Py_popcount32(uint32_t u)
+{
+#if defined(__clang__) || defined(__GNUC__)
+
+#if SIZEOF_INT == 4
+    return __builtin_popcount(u);
+#elif SIZEOF_LONG == 4
+    return __builtin_popcountl(u);
+#else
+#  error "unsupported configuration"
+#endif
+
+#else
+    /* 32-bit SWAR (SIMD Within A Register) popcount */
+    u -= (u >> 1) & UINT32_C(0x55555555);
```

Comment:
A minor point, but I prefer `(uint32_t)(0x55555555)`, because it involves less deduction to infer the actual type of the argument: the type is clearly a 32-bit unsigned int.

UINT32_C implies type `uint_least32_t`, which is "an unsigned integer type with a width of at least 32, such that no unsigned integer type with lesser size has at least width 32". And then as a reader of the code, you realise that

> Aha, but we _do_ have `uint32_t` available, so if `uint_least32_t` had width strictly larger than 32, then there would be an unsigned integer type with lesser size and at least width 32, which would contradict the definition of `uint_least32_t`. So `uint_least32_t` must have width exactly 32! QED.

I prefer not to have to go through that reasoning. :-)

Reply:
I added "const uint32_t ..." constants to be more explicit on used types and to ease reading.